extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageCapabilitiesUtil.java

```diff
@@ -32,16 +36,34 @@ public static Optional<String> getActiveContainerImageCapability(Capabilities ca
         if (activeContainerImageCapabilities.size() > 1) {
             throw new IllegalStateException(String.join(" and ", activeContainerImageCapabilities)
                     + " were detected, at most one container-image extension can be present.\n"
-                    + "Either remove the unneeded ones, or select one by adding the property 'quarkus.container-image.builder=<extension name (without the `container-image-` prefix)>' in application.properties or as a system property.");
+                    + "Either remove the unneeded ones, or select one by setting any of the following configuration properties: "
+                    + createBuilderSelectionPropertySuggestion(capabilities));
         }
         return activeContainerImageCapabilities.isEmpty() ? Optional.empty()
                 : Optional.of(activeContainerImageCapabilities.iterator().next());
     }
 
+    private static StringBuilder createBuilderSelectionPropertySuggestion(Capabilities capabilities) {
+        StringBuilder suggestion = new StringBuilder();
+        boolean isFirst = true;
+        for (String capability : capabilities.getCapabilities()) {
+            if (!isContainerImageCapability(capability)) {
+                continue;
+            }
+            if (!isFirst) {
+                suggestion.append(',');
```

Comment:
Micro comment: wouldn't it be a bit more readable with a space after the comma?

Reply:
Indeed, I'll update it